packages/typespec-test/test/todo_non_branded/generated/typespec-ts/src/api/todoItems/index.ts

```diff
@@ -0,0 +1,366 @@
+): StreamableMethod<
+  | TodoItemsCreate200Response
+  | TodoItemsCreate400Response
+  | TodoItemsCreate422Response
+  | TodoItemsCreate500Response
+> {
+  return context
+    .path("/items")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      contentType: (options.contentType as any) ?? "application/json",
+      body: {
+        item: {
+          title: item["title"],
+          assignedTo: item["assignedTo"],
+          description: item["description"],
+          status: item["status"],
+          labels: item["labels"],
+          _dummy: item["dummy"],
+        },
+        attachments: options?.attachments,
+      },
+    });
+}
+
+export async function _createDeserialize(
+  result:
+    | TodoItemsCreate200Response
+    | TodoItemsCreate400Response
+    | TodoItemsCreate422Response
+    | TodoItemsCreate500Response,
+): Promise<{
+  id: number;
+  title: string;
+  createdBy: number;
+  assignedTo?: number;
+  description?: string;
+  status: "NotStarted" | "InProgress" | "Completed";
+  createdAt: Date;
+  updatedAt: Date;
+  completedAt?: Date;
+  labels?: TodoLabels;
+}> {
+  if (result.status !== "200") {
+    throw createRestError(result);
+  }
+
+  return {
+    id: result.body["id"],
+    title: result.body["title"],
+    createdBy: result.body["createdBy"],
+    assignedTo: result.body["assignedTo"],
+    description: result.body["description"],
+    status: result.body["status"],
+    createdAt: new Date(result.body["createdAt"]),
+    updatedAt: new Date(result.body["updatedAt"]),
+    completedAt:
+      result.body["completedAt"] !== undefined
+        ? new Date(result.body["completedAt"])
+        : undefined,
+    labels: result.body["labels"],
+  };
+}
+
+export async function create(
+  context: Client,
+  item: TodoItem,
+  options: TodoItemsCreateOptionalParams = { requestOptions: {} },
+): Promise<{
+  id: number;
+  title: string;
+  createdBy: number;
+  assignedTo?: number;
+  description?: string;
+  status: "NotStarted" | "InProgress" | "Completed";
+  createdAt: Date;
+  updatedAt: Date;
+  completedAt?: Date;
+  labels?: TodoLabels;
+}> {
+  const result = await _createSend(context, item, options);
+  return _createDeserialize(result);
+}
+
+export function _getSend(
+  context: Client,
+  id: number,
+  options: TodoItemsGetOptionalParams = { requestOptions: {} },
+): StreamableMethod<TodoItemsGet200Response | TodoItemsGet404Response> {
+  return context
+    .path("/items/{id}", id)
+    .get({ ...operationOptionsToRequestParameters(options) });
+}
+
+export async function _getDeserialize(
+  result: TodoItemsGet200Response | TodoItemsGet404Response,
+): Promise<{
+  id: number;
+  title: string;
+  createdBy: number;
+  assignedTo?: number;
+  description?: string;
+  status: "NotStarted" | "InProgress" | "Completed";
+  createdAt: Date;
+  updatedAt: Date;
+  completedAt?: Date;
+  labels?: TodoLabels;
+}> {
+  if (result.status !== "200" && result.status !== "404") {
```

Comment:
I am clarifying the return type in issue: https://github.com/Azure/typespec-azure/issues/852